### PR TITLE
Final deploy

### PR DESCRIPTION
**Pull request by Mutai-Gilbert**
- [ ] In this pull request I was able to implement the following:

- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

Video presentation - https://www.loom.com/share/4a6e16f75e874b6480331e6cf8a3bba0?sid=d4b8e79d-0092-4e80-89b3-58a5f187a0a2

Live Project:- https://budget-nt3b.onrender.com

Test User:-
email: test@gmail.com
pass: test123

### CODE REVIEW

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Mutai, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🏁 🚦 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Properly added required models, controllers, and corresponding views
- Devise correctly implemented for authentication!
- Schema follow given ERD!
- App is running as intended!

## Required Changes ♻️
Your is working fine, good job! 👏 However, it will be a good idea if we can add a few tweaks to the design aspect so as to improve the look and feel. Please take note that it's one of the project's objectives that our design closely mimics the given [template design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=). Let's improve upon the design aspect by introducing the following tweaks:
- [x] Let's ensure that the blue header is present on all pages, fits, and should always stick to the top of the screen as seen in the original design.
- [x] Apparently, it's quite tedious navigating through your app. Let's kindly add a "Back" button (<) within the top blue navigation header section so as to enable users easily navigate back to the previous screen or to the home page.
- [x] Let's center the  "Create New Entity" text in the big green button below the page as seen from the [template design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) 
- [x] Let's improve the design of the green button below the page by adjusting the color to a lighter green color, and adding a white inner border within the button as seen in the original design.

 **Your categories page:**        |  **The original (similar page):**     
:-------------------------:|:-------------------------:
Home Page | Home Page

-  [x] Good job with the project presentation, however, you missed some steps in your video presentation, as mentioned in the previous review. Your video presentation only showed the feature aspects of your app, however, highlights of some portions of your code are missing in the presentation. Kindly ensure that the video presentation contains a highlight about a piece of the code (possibly from your code editor) that you find interesting or that you were proud of making.

_Kindly check the comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -275,14 +278,17 @@ button:hover {
...
+  padding: 28px;
   text-decoration: none;
   color: white;
   position: absolute;
   bottom: 0;
   left: 0;
-  border: none;
+  border: 2px solid white;
   right: 0;
 }
```
Comment (1GORDON) [inline app/assets/stylesheets/application.css]: Great job @Mutai-Gilbert 🤝

- [ ] Great job on centering this button just as the previous reviewer suggested. However, you forgot to add the inner white border just like the design appears in the design guidelines.
- Your design
your-button
- Given design
given-button

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -333,3 +339,11 @@ input[type="submit"] {
...
   margin-top: 10px;
 }
+
+.back-button {
+  color: grey;
+  text-decoration: none;
+  font-size: 15px;
+  margin: 2.5px;
+  border: 1px solid grey;
+}
```
Comment (1GORDON) [inline app/assets/stylesheets/application.css]: Great job @Mutai-Gilbert 🤝

- [x] Great job on adding this back button just like the previous reviewer suggested. However, kindly ensure this button can help in the navigation of your application. It is still very hard to navigate back to the home page.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Mutai-Gilbert ✋

Good job so far!

## Highlights!
- Project deployed on render ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [x] As stated by the previous reviewer, kindly remember to showcase a piece of code you are proud of building. You have only presented the main features of your project in the video but you forgot to showcase a piece of code you are proud of building.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @Mutai-Gilbert,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- Well-organized and clean project ✔️
- Professional readme file ✔️
- Correct workflow was used ✔️
- Good commit history ✔️

approv


# Status: Approved ✔️✔️✔️

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**